### PR TITLE
Replace several uses of `console.warn` with `warnOnce`

### PR DESCRIPTION
**Pull request by robertknight**
This reduces warning spam in the event where the code path that triggers
the warning is invoked repeatedly and there is no value to the developer
in telling them about the issue more than once.

To make `warnOnce` an easier drop-in replacement for `console.warn`, it
has been modified to accept multiple arguments, all of which get
forwarded to `console.warn`.

### CODE REVIEW

File: src/shared/warn-once.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+'use strict';
```
Comment (robertknight) [inline src/shared/warn-once.js]: This file has been moved from `src/sidebar/util/warn-once.js` and modified slightly to make it a better drop-in replacement for `console.warn`.

File: src/annotator/features.js [hunk 2]
```diff
@@ -19,7 +20,7 @@ module.exports = {
 
   flagEnabled: function(flag) {
     if (!(flag in _features)) {
-      console.warn('looked up unknown feature', flag);
+      warnOnce('looked up unknown feature', flag);
```
Comment (lyzadanger) [inline src/annotator/features.js]: This is so great!

File: src/annotator/test/features-test.js [hunk 3]
```diff
@@ -51,10 +56,10 @@ describe('features - annotation layer', function() {
     });
 
     it('should warn when accessing unknown flags', function() {
-      assert.notCalled(console.warn);
+      assert.notCalled(fakeWarnOnce);
```
Comment (lyzadanger) [inline src/annotator/test/features-test.js]: Given that the `sinon.stub` is re-initialized before each test, is this test useful?

File: src/annotator/test/features-test.js [hunk 4]
```diff
@@ -51,10 +56,10 @@ describe('features - annotation layer', function() {
...
       assert.isFalse(features.flagEnabled('unknown_feature'));
-      assert.calledOnce(console.warn);
-      assert.calledWith(console.warn, 'looked up unknown feature');
+      assert.calledOnce(fakeWarnOnce);
```
Comment (lyzadanger) [inline src/annotator/test/features-test.js]: Perhaps use `calledOnceWith` here?

File: src/sidebar/services/features.js [hunk 5]
```diff
@@ -13,9 +13,10 @@
 
 const events = require('../events');
 const bridgeEvents = require('../../shared/bridge-events');
+const warnOnce = require('../../shared/warn-once');
 
 // @ngInject
-function features($log, $rootScope, bridge, session) {
+function features($rootScope, bridge, session) {
```
Comment (lyzadanger) [inline src/sidebar/services/features.js]: I like the reduction in function args here.

File: src/annotator/test/features-test.js (hunk 3 above)
Comment (robertknight) [reply src/annotator/test/features-test.js]: Good catch!

File: src/annotator/test/features-test.js (hunk 4 above)
Comment (robertknight) [reply src/annotator/test/features-test.js]: Oh wait ... it turns out there is no `calledOnceWith` function. That only exists in Python-land (as `called_once_with`)

Comment (lyzadanger) [reply src/annotator/test/features-test.js]: Are we not using the chai plugin for `sinon`? https://www.chaijs.com/plugins/sinon-chai/

Comment (robertknight) [reply src/annotator/test/features-test.js]: Ah, we're just using Sinon's [`assert.expose` function](https://sinonjs.org/releases/v7.3.2/assertions/) -  see https://github.com/robertknight/client/blob/ac91921b940ce23bf5b1981de8482f2471aff3ef/src/sidebar/test/bootstrap.js#L4

Depending on what else that plugin gives us, maybe we should consider it.

**Pull request merged**